### PR TITLE
Diagnostic for barotropic velocity tendency

### PR DESCRIPTION
**Pull request by hmkhatri**
I have updated the MOM_barotropic.F90 file to save barotropic velocity tendency as a diagnostic. The term is computed by taking the difference between barotropic velocities at the end and the start of a baroclinic timestep and dividing by the baroclinic timestep.

### CODE REVIEW

File: src/core/MOM_barotropic.F90 [hunk 1]
```diff
@@ -2300,6 +2313,19 @@ subroutine btstep(U_in, V_in, eta_in, dt, bc_accel_u, bc_accel_v, forces, pbce,
       enddo ; enddo
       call post_data(CS%id_Corv_bt, Corv_bt_sum(isd:ied,JsdB:JedB), CS%diag)
     endif
+    if (CS%id_ubtdt > 0) then
+      do j=js,je ; do I=is-1,ie
+        ubt_dt(I,j) = (ubt_wtd(I,j) - ubt_st(I,j))/dt
```
Comment (ashao) [inline src/core/MOM_barotropic.F90:2318]: Division is a little more expensive than multiplication, so in other parts of the code, the inverse is often calculated and stored near the beginning of the routine (i.e. `Idt = 1./dt`). This expression would then change to:
```
ubt_dt(I,j) = (ubt_wtd(I,j) - ubt_st(I,j))*Idt
```
Most compilers catch this during optimization, but it doesn't really hurt to add it explicitly.

Comment (hmkhatri) [reply src/core/MOM_barotropic.F90:2318]: Thanks. I will make this change.

**Comment by marshallward** [comment]
I've confirmed that the only regression is due to the introduction of the `[uv]bt_dt` diagnostics:
```
385,388d384
< u-point: mean=  -4.6736044829326117E-08 min=  -2.6143208716579239E-07 max=   0.0000000000000000E+00 ocean_model-ubt_dt
< u-point: c=       958 ocean_model-ubt_dt
< v-point: mean=   1.2619092135694276E-06 min=   0.0000000000000000E+00 max=   7.1769065749685773E-06 ocean_model-vbt_dt
< v-point: c=       782 ocean_model-vbt_dt
```

**Comment by adcroft** [comment]
Is the PR being updated or taken as is?

**Comment by Hallberg-NOAA** [comment]
The question of whether to take this as is or to do the minor suggested update is up to @hmkhatri . The suggestions for code modification should minorly improve performance and reduce the memory footprint, but will not alter the answers or diagnostics.

**Comment by hmkhatri** [comment]
I am in the process of implementing the suggestions made by @Hallberg-NOAA.  I will push the modifications to the branch soon.

**Comment by hmkhatri** [comment]
Suggestions by @Hallberg-NOAA have been implemented.

File: src/core/MOM_barotropic.F90 [hunk 2]
```diff
@@ -1557,7 +1557,6 @@ subroutine btstep(U_in, V_in, eta_in, dt, bc_accel_u, bc_accel_v, forces, pbce,
       ubt_st(I,j) = ubt(I,j) 
     enddo ; enddo
   endif
-
   if (CS%id_vbtdt > 0) then
```
Comment (Hallberg-NOAA) [inline src/core/MOM_barotropic.F90:1560]: This change removed a blank line, but the fix that actually was needed was to remove the spaces at the end of the two lines a few lines above.  Completely blank lines (without any space or tabs) are fine, but it is a MOM6 convention that there are no spaces after the last non-space character on any line.  To find trailing white space, try the command `grep -n ' $' src/*/*.F90 src/parameterizations/*/*.F90 config_src/*/*.F90` when sitting in your MOM6 directory.

**Pull request merged**